includes/Proxies/Tracks.php

```diff
@@ -0,0 +1,36 @@
+<?php
+declare( strict_types=1 );
+
+namespace SkyVerge\WooCommerce\Facebook\Proxies;
+
+use WC_Site_Tracking;
+use WC_Tracks;
+
+/**
+ * Wrapper (proxy) class for WC_Tracks from WooCommerce Core.
+ *
+ * Supplies a standard prefix to all events.
+ *
+ * @package Automattic\WooCommerce\Facebook\Proxies
+ */
+class Tracks {
+
+	/**
+	 * Record a tracks event.
+	 *
+	 * @param string $name       The event name to record.
+	 * @param array  $properties Array of properties to include with the event.
+	 */
+	public static function record_event( string $name, array $properties = [] ) {
+		if ( ! class_exists( WC_Tracks::class ) || ! WC_Site_Tracking::is_tracking_enabled() ) {
```

Comment:
Looks like we are doubling up by checking `WC_Site_Tracking::is_tracking_enabled()` here.

https://github.com/woocommerce/woocommerce/blob/dc6b44fc6f733e0526c5e16786ac9715aa019b74/includes/tracks/class-wc-tracks.php#L80-L82

Reply:
Good point - I was inspired by the implementation in another plugin (pasted the proxy class over and renamed) 😁 

> I really like how you set this up following the proxy pattern! I think it would be great to make the `record_event` method non-static. That way we can mock calls to it in any future unit tests.

I don't see a huge benefit to mock this class in tests, but I think it's a good idea to do avoid `static` just for consistency. The main concern I have is that the utility is easy to use in client code, and doesn't affect the interface of client code. Should be as close to a one-liner drop in when you want to add an event somewhere in PHP.

Reply:
> I don't see a huge benefit to mock this class in tests

Mocking the class would be a neat way to test that a specific event was recorded without actually recording the event. Or just to be certain about avoiding executing tracking related code on a test site.

> Should be as close to a one-liner drop in when you want to add an event somewhere in PHP.

Definitely. Since we don't have a DI container I'd suggest something like: `facebook_for_woocommerce()->tracks_proxy()->record_event()`